controllers/chat/ChatController.js

```diff
@@ -160,39 +186,140 @@ module.exports = {
+        anon_user_info_emoji_code: 'string|empty:false',
+        anon_user_info_emoji_name: 'string|empty:false',
+      };
+      const validated = v.validate(req.body, schema);
+      if (validated.length)
+        return res.status(403).json({
+          message: 'Error validation',
+          error: validated,
+        });
+      const {
+        members,
+        message,
+        anon_user_info_color_code,
+        anon_user_info_color_name,
+        anon_user_info_emoji_code,
+        anon_user_info_emoji_name,
+      } = req.body;
+      if (!members.includes(req.userId)) members.push(req.userId);
+
+      const client = StreamChat.getInstance(
+        process.env.API_KEY,
+        process.env.SECRET
+      );
+
+      await client.connectUser({ id: req.userId }, req.token);
+
+      const channel = client.channel('messaging', { members });
+
+      await channel.create();
+
+      const chat = await channel.sendMessage({
+        user_id: req.userId,
+        text: message,
+        ...req.body,
+      });
+
+      const targets = members.filter((member) => member !== req.userId);
+      targets.map(async (target) => {
+        await ChatAnonUserInfo.create({
+          channelId: channel.id,
+          myAnonUserId: req.userId,
+          targetUserId: target,
+          anon_user_info_color_code,
+          anon_user_info_color_name,
+          anon_user_info_emoji_code,
+          anon_user_info_emoji_name,
+        });
+      });
+      await saveChat({ user_id: req.userId, channelId: channel.id, message });
+
+      await client.disconnectUser();
+
+      return res.status(200).json(responseSuccess('sent', chat));
+    } catch (error) {
+      return ErrorResponse.e400(res, error.message);
+    }
+  },
+  getMyAnonProfile: async (req, res) => {
     const schema = {
-      members: 'string[]|empty:false',
-      message: 'string|empty:false',
+      channelId: 'string',
     };
     const validated = v.validate(req.body, schema);
     if (validated.length)
       return res.status(403).json({
         message: 'Error validation',
         error: validated,
       });
-    if (!req.body.members.includes(req.userId))
-      req.body.members.push(req.userId);
 
-    const client = StreamChat.getInstance(
-      process.env.API_KEY,
-      process.env.SECRET
+    const myAnonUserId = await UsersFunction.findAnonymousUserId(
+      User,
+      req.userId
     );
-
-    await client.connectUser({ id: req.userId }, req.token);
-
-    const channel = client.channel('messaging', { members: req.body.members });
-
-    await channel.create();
-
-    const message = await channel.sendMessage({
-      user_id: req.userId,
-      text: req.body.message,
-      ...req.body,
+    const { channelId } = req.body;
+    const haveChat = await ChatAnonUserInfo.findOne({
+      where: {
+        channelId,
+        targetUserId: req.params.userId,
+        myAnonUserId: myAnonUserId.user_id,
+      },
```

Comment:
You should query channel Id before checking to the DB

Reply:
I will check the anon user info saving process to DB after this change requests are fixed